lib/python/picongpu/picmi/species.py

```diff
@@ -1,54 +1,103 @@
 """
 This file is part of PIConGPU.
-Copyright 2021-2023 PIConGPU contributors
+Copyright 2021-2024 PIConGPU contributors
 Authors: Hannes Troepgen, Brian Edward Marre
 License: GPLv3+
 """
 
-from ..pypicongpu import util, species
+from .. import pypicongpu
+from ..pypicongpu.species.util.element import Element
+from .interaction import InteractionInterface
+from .predefinedparticletypeproperties import non_element_particle_type_properties, _PropertyTuple
 
 import picmistandard
 
-import typeguard
 import typing
+import typeguard
+import pydantic
+import pydantic_core
 import logging
+import re
+
 from scipy import constants as consts
 
 
 @typeguard.typechecked
 class Species(picmistandard.PICMI_Species):
     """PICMI object for a (single) particle species"""
 
-    # ONLY set non-element particles here -- all other are handled by
-    # element
-    __mass_charge_by_openpmd_name_non_elements = {
-        "electron": (consts.electron_mass, -consts.elementary_charge),
-    }
-    """mass/charge tuple to use when passed a non-element particle_type"""
-
-    picongpu_fully_ionized = util.build_typesafe_property(typing.Optional[bool])
+    __non_element_particle_type_properties = non_element_particle_type_properties
     """
-    *usually* ionization is expected to be used on elements -- use this to
-    explicitly DISABLE ionization
+    mass/charge to use when passed a non-element particle_type
+
+    @attention ONLY set non-element particles here, all other are handled by element
     """
 
-    def __init__(self, picongpu_fully_ionized: typing.Optional[bool] = None, picongpu_ionization_electrons=None, **kw):
-        self.picongpu_fully_ionized = picongpu_fully_ionized
+    __non_element_particle_types: dict[str, _PropertyTuple] = __non_element_particle_type_properties.keys()
```

Comment:
That doesn't sound about right. Maybe this instead?
```suggestion
    __non_element_particle_types: dict[str, _PropertyTuple] = __non_element_particle_type_properties
```

Reply:
fixed

Reply:
they type annotation was wrong it should have been `list[str]`